src/borg/cache.py

```diff
@@ -911,7 +912,8 @@ def file_known_and_unchanged(self, path_hash, st, ignore_inode=False):
         if not entry:
             return None
         entry = FileCacheEntry(*msgpack.unpackb(entry))
-        if (entry.size == st.st_size and bigint_to_int(entry.mtime) == st.st_mtime_ns and
+        cmtime = st.st_mtime_ns if use_mtime else st.st_ctime_ns
+        if (entry.size == st.st_size and bigint_to_int(entry.cmtime) == cmtime and
```

Comment:
interesting detail:

as we use `==`, it does not matter whether the fs timestamp is newer or older than the cached timestamp.

guess that means switching from mtime to ctime or vice versa does not create any harmful effect (besides hashing some stuff again), so we do not need `borg delete --cache-only` when switching.